### PR TITLE
Support java.nio.Path objects via API

### PR DESCRIPTION
**Pull request by RickMoynihan**

### CODE REVIEW

File: test/csv2rdf/csvw_test.clj [hunk 1]
```diff
@@ -6,28 +6,37 @@
...
-    (is (= 0 (count warnings)) "Received warnings but none was expected")
-    (is (= 0 (count errors)) "Received errors but none was expected")))
+  (:import [java.net URI]
+           [java.nio.file FileSystems]
+           [grafter_2.rdf.protocols Quad]))
+
+(defn- make-path [& path]
+  (.getPath (FileSystems/getDefault) "w3c-csvw" (into-array String path)))
+
+(deftest csv->rdf-with-paths
```
Comment (lkitching) [inline test/csv2rdf/csvw_test.clj:16]: Could we move this test into a different file - the `csvw_test.clj` file is generated from the test `manifest.csv` file. See `csvw-test.gen/write-tests-file`.

File: src/csv2rdf/csvw.clj [hunk 2]
```diff
@@ -19,11 +19,42 @@
     (table-statements context table annotated-rows)))
 
 (defn csv->rdf
-  "Runs the CSVW process for the given tabular or metadata data sources and options. If metadata-source
-   is non-nil then processing will start from the asscociated metadata document, otherwise it will start
-   from tabular-source. Returns a lazy sequence of statements containing the CSVW output for the specified
-   CSVW mode. Mode can be specified by the :mode key of the options map if provided, otherwise standard mode
-   will be used."
+  "Runs the CSVW process for the given tabular or metadata data sources
```
Comment (lkitching) [inline src/csv2rdf/csvw.clj:22]: Big improvement, thanks!

File: src/csv2rdf/source.clj [hunk 3]
```diff
@@ -58,7 +63,11 @@
   (get-json [f] (read-json f))
 
   String
-  (get-json [s] (json/read-str s)))
+  (get-json [s] (json/read-str s))
+
+  Path
+  (get-json [p]
+    (get-json (slurp (io/reader (Files/newInputStream p (into-array OpenOption [StandardOpenOption/READ])))))))
```
Comment (lkitching) [inline src/csv2rdf/source.clj:70]: Would `(read-json (Files/newInputStream p (into-array OpenOption [StandardOpenOption/READ])))` also work here?

**Review by lkitching — CHANGES_REQUESTED** [review]
This mostly looks good apart from the new test location.

File: test/csv2rdf/csvw_test.clj (hunk 1 above)
Comment (RickMoynihan) [reply test/csv2rdf/csvw_test.clj:16]: yeah I did consider putting it in a different file, as I knew they'd been generated... just wasn't sure exactly how.

File: src/csv2rdf/csvw.clj (hunk 2 above)
Comment (RickMoynihan) [reply src/csv2rdf/csvw.clj:22]: You're welcome 🙇

File: src/csv2rdf/source.clj (hunk 3 above)
Comment (RickMoynihan) [reply src/csv2rdf/source.clj:70]: Oh thanks I meant to check that, but forgot; and yes it works.

Comment (RickMoynihan) [reply src/csv2rdf/source.clj]: Fixed and forced pushed

**Comment by RickMoynihan** [comment]
Ok,

I'll create a new PR to move the test suite and update the generator accordingly; and then rebase this ontop of it.

**Pull request merged**